src/Stratis.Bitcoin/P2P/PeerAddressManager.cs

```diff
@@ -0,0 +1,313 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using NBitcoin;
+using NBitcoin.Protocol;
+using Stratis.Bitcoin.Utilities;
+using Stratis.Bitcoin.Utilities.FileStorage;
+
+namespace Stratis.Bitcoin.P2P
+{
+    /// <summary>
+    /// The AddressManager keeps a set of peers discovered on the network in cache and on disk.
+    /// <para>
+    /// The manager updates their states according to how recent they have been connected to.
+    /// </para>
+    /// </summary>
+    public sealed class PeerAddressManager : IDisposable
+    {
+        private IAsyncLoop asyncLoop;
+        private readonly IAsyncLoopFactory asyncLoopFactory;
+        private readonly string peerFilePath;
+        internal const string PEERFILENAME = "peers.json";
+
+        public PeerAddressManager(IAsyncLoopFactory asyncLoopFactory, string peerFilePath)
```

Comment:
If you replace the peerFilePath with the DataFolder class you will be able to just inject this class in the feature.

Reply:
Fixed